collector/src/bin/rustc-perf-collector/execute.rs

```diff
@@ -102,13 +102,16 @@ struct CargoProcess<'sysroot> {
 #[derive(Copy, Clone, Debug)]
 struct Incremental(bool);
 
-impl<'sysroot> CargoProcess<'sysroot> {
+impl<'a> CargoProcess<'a> {
     fn base_command(&self, cwd: &Path, subcommand: &str) -> Command {
-        let mut cmd = self.sysroot.command("cargo");
+        let mut cmd = Command::new(Path::new("cargo"));
         cmd
-            .env("SHELL", env::var_os("SHELL").unwrap_or_default())
```

Comment:
I've had problems with some benchmarks in the past if the SHELL environment variable isn't passed down, but if this works, I'm fine with landing it...

Reply:
To be fair, I've only tested `helloworld` and `parser`! I will run all the benchmarks to see if SHELL or anything else is needed.